Sofa/framework/Core/src/sofa/core/objectmodel/BaseObject.cpp

```diff
@@ -342,22 +342,19 @@ SReal BaseObject::getTime() const
     return getContext()->getTime();
 }
 
-std::string BaseObject::getPathName() const {
-
-    const BaseContext* context = this->getContext();
-    std::string result = "";
-    if( context )
-    {
-        const BaseNode* node = context->toBaseNode();
-        if( node ) {
-            result += node->getPathName();
-            if (node->getPathName() != "/")
-                result += "/";
-        }
-
-    }
-    result += getName();
-    return result;
+std::string BaseObject::getPathName() const
+{
+    auto node = dynamic_cast<const BaseNode*>(getContext());
+    if(!node)
+        return "";
+
+    auto pathname = node->getPathName();
+    std::stringstream tmp;
+    tmp << pathname;
+    if (pathname != "/")
+        tmp << "/";
+    tmp << getName();
+    return tmp.str();
```

Comment:
```suggestion
    return sofa::helper::system::FileSystem::append(node->getPathName(), getName());
```
Not really a file, but it does the job

Reply:
ouch i breaks my eyes... I wouldn't do such a thing unless in <sofa/helper/stringutils.h> there is something like pathclean. I let other decide :)